tests/queries/0_stateless/01948_heredoc.sql

```diff
@@ -0,0 +1,4 @@
+SELECT $$VALUE$$;
```

Comment:
Need a test with multiline values.
And also with dollar-something sequences in the middle.

$doc$

$do$
$ doc$
$doc $
$doco$

$doc$;

Reply:
Also test with cyrillic/chinese characters - both for separators and for values.
A test with non-UTF8 binary data inside doc will be also nice to have.

Reply:
Also add .sh test to check how server reacts to unfinished heredocs.
Both unfinished separator and unfinished content. Zero and non-zero size.

Reply:
Fixed.